Mathlib/Topology/Algebra/Module/PerfectPairing.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2025 Yaël Dillies, Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Andrew Yang
+-/
+import Mathlib.LinearAlgebra.BilinearMap
+import Mathlib.Topology.Algebra.Module.LinearMap
+
+/-!
+# Continuous perfect pairings
+
+This file defines continuous perfect pairings.
+
+For a topological ring `R` and two topological modules `M` and `N`, a continuous perfect pairing is
+a continuous bilinear map `M × N → R` that is bijective in both arguments.
+
+We require continuity in the forward direction only so that we can put several different topologies
+on the continuous dual: strong, weak, weak-* topology...
```

Comment:
How about we give a bit more detail here:
```suggestion
We require continuity in the forward direction only since we do not wish to commit to a choice of
topology on the continuous dual (e.g., strong, weak, weak-*). For example, by weak reflexivity there
is a continuous perfect pairing between `M` and `WeakDual R M`, even though the map
`M ≃ₗ[R] (WeakDual R M →L[R] R)` is not in general a homeomorphism.
```